### PR TITLE
Add current and defaultScopes when refreshing session

### PR DESCRIPTION
**Pull request by szubster**
Otherwise google auth always fails, as it lacks userinfo.profile

## Hey, I just made a Pull Request!

Was fighting with k8s plugin + google auth.
It turned out, that when refreshing session only explicitly provided scopes are taken into account.
I do not know whether this is google's quirk, or a standard.
Without this, user needs to provide auth information on every page refresh.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by szubster** [comment]
I know tests are failing, but I do not know whether I should try to fix Microsoft Auth, or is the problem I am trying to solve google-specific.

**Comment by szubster** [comment]
I have also tried using `include_granted_scopes` param from passport-google-oauth2 library, but then I cannot pass scopes at all when refreshing token

**Comment by szubster** [comment]
ok, it seems that if you use refresh token with scopes specified it returns access token with only those scopes.
during refresh backstage always [fetches profile](https://github.com/backstage/backstage/blob/master/plugins/auth-node/src/oauth/PassportOAuthAuthenticatorHelper.ts#L116) but the scopes of returned token only contain explicitly requested scopes.

My proposals is google auth can either always add profile and email scopes (only this authenticator) or in google's auth stop passing scopes and add `include_granted_scopes` option.

@freben any advice?

**Comment by szubster** [comment]
bump @freben

**Comment by szubster** [comment]
replaced by: **https://github.com/backstage/backstage/pull/20735**

File: packages/core-app-api/src/apis/implementations/auth/microsoft/MicrosoftAuth.test.ts [hunk 1]
```diff
@@ -59,15 +57,17 @@ describe('MicrosoftAuth', () => {
     it('gets access token with requested scopes for grant', async () => {
       const accessToken = await microsoftAuth.getAccessToken();
 
-      expect(accessToken).toEqual('tokenForGrantScopes');
+      expect(accessToken).toEqual(
+        'token:openid offline_access profile email User.Read',
```
Comment (Rugvip) [inline packages/core-app-api/src/apis/implementations/auth/microsoft/MicrosoftAuth.test.ts:61]: @jamieklassen 👀 🙏

**Review by Rugvip — COMMENTED** [review]
Reopened and pushed some test fixes

**Review by Rugvip — APPROVED** [review]
Feeling fairly confident in this fix so gonna ship it through. It should not impact the Microsoft auth instance split as the additional instances do not include default scopes: https://github.com/backstage/backstage/blob/56baf8763ad4840351a4a42fbb6b49645e0815ea/packages/core-app-api/src/apis/implementations/auth/microsoft/MicrosoftAuth.ts#L146-L153

Thank you @szubster 👍

**Pull request merged**